### ISSUE TITLE
perlipc code example with exec inside signal handler

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13352]
Migrated from [rt.perl.org#120256](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=120256) (status was 'open')

Searchable as RT120256$

**Comment by p5pRT** [comment]
### From victor@vsespb.ru

This example is broken&#8203;:
http&#8203;://perldoc\.perl\.org/perlipc\.html\#Handling\-the\-SIGHUP\-Signal\-in\-Daemons

it only works for first SIGHUP\.

looks like it's because this issue
http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2000/01/msg5237\.html
more investagations and workarounds here&#8203;:
http&#8203;://www\.perlmonks\.org/?node\_id=1058565

I can fix the example by setting flag in signal handler\, and calling exec
outside of handler\, but probably there are better fixes with
POSIX&#8203;::SigAction etc which I am not familar with\.

**Comment by p5pRT** [comment]
### From @Leont

On Thu\, Oct 17\, 2013 at 7&#8203;:29 PM\, Victor Efimov \<perlbug\-followup@&#8203;perl\.org>wrote&#8203;:

> \# New Ticket Created by  Victor Efimov
> \# Please include the string&#8203;:  \[perl \#120256\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=120256 >
>
>
> This example is broken&#8203;:
> http&#8203;://perldoc\.perl\.org/perlipc\.html\#Handling\-the\-SIGHUP\-Signal\-in\-Daemons
>
> it only works for first SIGHUP\.
>
> looks like it's because this issue
> http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2000/01/msg5237\.html
> more investagations and workarounds here&#8203;:
> http&#8203;://www\.perlmonks\.org/?node\_id=1058565
>
> I can fix the example by setting flag in signal handler\, and calling exec
> outside of handler\, but probably there are better fixes with
> POSIX&#8203;::SigAction etc which I am not familar with\.
>
>
> This example is broken&#8203;:
> http&#8203;://perldoc\.perl\.org/perlipc\.html\#Handling\-the\-SIGHUP\-Signal\-in\-Daemons
>
> it only works for first SIGHUP\.
>
> looks like it's because this issue
> http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2000/01/msg5237\.html
> more investagations and workarounds here&#8203;:
> http&#8203;://www\.perlmonks\.org/?node\_id=1058565
>
> I can fix the example by setting flag in signal handler\, and calling exec
> outside of handler\, but probably there are better fixes with
> POSIX&#8203;::SigAction etc which I am not familar with\.
>

Right\, this makes sense\.

During the signal handler\, the signal is temporarily blocked\. On exec\, the
new process inherits this block\.

Actually the proposed fix sucks too \(it uses an unsafe signal handler\);
possibly a better approach is to unblock it before the exec call\.

Leon

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'
